datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -617,12 +619,29 @@ protected void handlePollRecordsException(Exception e) throws Exception {
       _logger.warn("Poll fail with exception", e);
       throw e;
     } else {
-      _logger.warn(
-          String.format("Poll threw an exception. Sleeping for %d seconds and repoll from consumer, poll attempt: %d",
-              _retrySleepDuration.getSeconds(), _pollAttempts.intValue()), e);
-    }
-    if (!_shutdown) {
-      Thread.sleep(_retrySleepDuration.toMillis());
```

Comment:
why are we removing this sleep behavior before next retry?

Reply:
that sleep of 5 seconds does not make sense. Ideally we need to have an exponential backoff retry. Here, by simply having 5 seconds sleep on any failure, can cause the lag to increase for other partitions which are assigned on the same task. Also, with this latest change, where we auto pause on source error, the sleep becomes not important.

Reply:
makes sense